lib/javascript.d.ts

```diff
@@ -195,10 +200,10 @@ declare global {
 		type Object = StateObject | ChannelObject | DeviceObject | OtherObject;
 		type PartialObject = PartialStateObject | PartialChannelObject | PartialDeviceObject | PartialOtherObject;
 
-		type GetStateCallback = (err: string | null, state?: State) => void;
-		type SetStateCallback = (err: string | null, id?: string) => void;
+		type GetStateCallback<T = any> = (err: string | null, state?: State<T>) => void;
```

Comment:
I'm torn on this. You can achieve the same by just asserting the type on the callback parameter, e.g.:
```ts
getState("id", (err, state?: iobJS.State<boolean>) => { /* ... */ });
```
This way the generic is more directly associated with the parameter it belongs to.

But I can see the benefit of just writing `getState<boolean>(...)` and not caring about the parameter type names. So let's do this.

Reply:
Regarding #248 I think the callback parameter should be `State<T> | AbsentState`, shouldn't it?